omod/src/main/java/org/openmrs/module/fhir2/web/servlet/FhirRestServlet.java

```diff
@@ -62,8 +63,19 @@ protected void initialize() {
 		setDefaultResponseEncoding(EncodingEnum.JSON);
 		registerInterceptor(loggingInterceptor);
 		
-		getFhirContext().setNarrativeGenerator(new CustomThymeleafNarrativeGenerator(
-		        FhirConstants.HAPI_NARRATIVES_PROPERTY_FILE, FhirConstants.OPENMRS_NARRATIVES_PROPERTY_FILE));
+		String narrativesOverridePropertyFile = NarrativeUtils.getValidatedPropertiesFilePath(
+		    globalPropertyService.getGlobalProperty(FhirConstants.NARRATIVES_OVERRIDE_PROPERTY_FILE, ""));
+		
+		String[] narrativePropertiesFiles;
+		if (narrativesOverridePropertyFile != null) {
+			narrativePropertiesFiles = new String[] { narrativesOverridePropertyFile,
+			        FhirConstants.HAPI_NARRATIVES_PROPERTY_FILE, FhirConstants.OPENMRS_NARRATIVES_PROPERTY_FILE };
+		} else {
+			narrativePropertiesFiles = new String[] { FhirConstants.HAPI_NARRATIVES_PROPERTY_FILE,
+			        FhirConstants.OPENMRS_NARRATIVES_PROPERTY_FILE };
+		}
```

Comment:
I've restructured this the way suggested [here](https://github.com/openmrs/openmrs-module-fhir2/pull/252#discussion_r455036175).